pkg/provisioner/controller.go

```diff
@@ -606,25 +606,25 @@ func (c *obcController) deleteResources(ob *v1alpha1.ObjectBucket, cm *corev1.Co
 }
 
 // Add finalizer and labels to the OBC.
-func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (err error) {
+func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (*v1alpha1.ObjectBucketClaim, error) {
 	clib := c.libClientset
 
 	logD.Info("getting OBC to set metadata fields")
-	obc, err = clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
+	obcFetched, err := clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("error getting obc: %v", err)
+		return nil, fmt.Errorf("error getting obc: %v", err)
 	}
 
-	obc.SetFinalizers([]string{finalizer})
-	obc.SetLabels(c.provisionerLabels)
+	addFinalizers(obcFetched, []string{finalizer})
+	addLabels(obcFetched, c.provisionerLabels)
 
 	logD.Info("updating OBC metadata")
-	obc, err = updateClaim(clib, obc)
+	obcUpdated, err := updateClaim(clib, obcFetched)
 	if err != nil {
-		return fmt.Errorf("error configuring obc metadata: %v", err)
+		return nil, fmt.Errorf("error configuring obc metadata: %v", err)
```

Comment:
Probably good to return the original OBC in case code that follows needs to process it in an error state  or in case the code in a future time attempts to continue on after an error.

```suggestion
		return obc, fmt.Errorf("error configuring obc metadata: %v", err)
```

Reply:
I will follow the existing error handling way to make the code style consistent, but if we wanna refer to the original obc value out of the invoked function, it's better to use the input value that was given to the function, IMO.
Besides, we might be confused if we go on using a value that is returned together with an error and if we wanna do that, we just should separate the function at the point the function returns a meaningful value.

Reply:
I don't think there is anything wrong with returning a value under an error condition in this case. This pattern is used elsewhere in the resourcehandlers. Consider pseudocode below where cleanUp() might need info from value.
```
value = some starting value
value, err = mutateValue(value)
if err {
  cleanUp(value)
  return err
}

```

Reply:
I tend to agree because, in general, code check for `err != nil` before it processes other func returns...